core/src/main/scala/org/apache/spark/scheduler/cluster/mesos/MesosClusterScheduler.scala

```diff
@@ -188,10 +188,10 @@ private[spark] class MesosClusterScheduler(
         mesosDriver.killTask(task.taskId)
         k.success = true
         k.message = "Killing running driver"
-      } else if (removeFromQueuedDrivers(submissionId)) {
```

Comment:
We should just rename and change the existing function, I Don't think it's being used elsewhere?